src/Microsoft.DotNet.Darc/src/Darc/Operations/UpdateDependenciesOperation.cs

```diff
@@ -121,61 +128,69 @@ public override async Task<int> ExecuteAsync()
 
                     foreach (string repoToQuery in repositoryUrisForQuery)
                     {
+                        Console.WriteLine($"Looking up latest build of {repoToQuery} on {_options.Channel}");
                         var latestBuild = barOnlyRemote.GetLatestBuildAsync(repoToQuery, channelInfo.Id.Value);
                         getLatestBuildTaskDictionary.TryAdd(repoToQuery, latestBuild);
                     }
 
-                    // Now walk dependencies again and attempt the update
-                    foreach (DependencyDetail dependency in dependencies)
+                    // For each build, first go through and determine the required updates,
+                    // updating the "live" dependency information as we go.
+                    // Then run a second pass where we update any assets based on coherency information.
+                    foreach (KeyValuePair<string, Task<Build>> buildKvPair in getLatestBuildTaskDictionary)
                     {
-                        Build build = await getLatestBuildTaskDictionary[dependency.RepoUri];
-                        
+                        string repoUri = buildKvPair.Key;
+                        Build build = await buildKvPair.Value;
                         if (build == null)
                         {
-                            Logger.LogTrace($"No build of '{dependency.RepoUri}' found on channel '{_options.Channel}'.");
+                            Logger.LogTrace($"No build of '{repoUri}' found on channel '{_options.Channel}'.");
                             continue;
                         }
+                        IEnumerable<AssetData> assetData = build.Assets.Select(
+                            a => new AssetData
+                            {
+                                Name = a.Name,
+                                Version = a.Version
+                            });
+                        
+                        // Now determine what needs to be updated.
+                        Dictionary<DependencyDetail, DependencyDetail> updates =
```

Comment:
>Dictionary [](start = 24, length = 10)

Is this using the dictionary as just a list of tuples? Should it just be a List<(before,after)>?

Reply:
I don't like tuples for widely used methods.  I chagned this over to using a separate DependencyUpdate type.